### PR TITLE
Add counter examples in Solidity and Rust

### PR DESCRIPTION
**Pull request by armaniferrante**

### CODE REVIEW

File: test/test-counters.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const Counter = artifacts.require("Counter");
+const NestedCounter = artifacts.require("NestedCounter");
+const ConfidentialCounter = artifacts.require("ConfidentialCounter");
+const WasmCounter = artifacts.requir("WasmCounter");
+const NestedWasmCounter = artifacts.requir("NestedWasmCounter");
+const ConfidentialWasmCounter = artifacts.requir("ConfidentialWasmCounter");
```
Comment (daviddao) [inline test/test-counters.js]: Small typo: `artifacts.requir` => `artifacts.require()`

**Review by daviddao — CHANGES_REQUESTED** [review]

File: test/test-counters.js (hunk 1 above)
Comment (armaniferrante) [reply test/test-counters.js]: Thanks :) 

Waiting for us to publish the latest owasm-abi-derive on crates.io before coming back to this.

Comment (armaniferrante) [reply test/test-counters.js]: Fixed

**Review by vishwa-raman — APPROVED** [review]
Tested against devnet using the latest c-k image. It works! Here are the results -

Post `truffle migrate --network oasis`
```
Summary
=======
> Total deployments:   7
> Final cost:          0.03414514 ETH
```

Post `truffle test --network oasis`
```
  Contract: Counter Contracts
    ✓ should have a starting count of 0 (261ms)
    ✓ should increment the count to 1 (1140ms)
    ✓ should have a starting count of 0 (62ms)
    ✓ should increment the count to 1 (870ms)
    ✓ should have a starting count of 0 (607ms)
    ✓ should increment the count to 1 (1024ms)
    ✓ should have a starting count of 0 (106ms)
    ✓ should increment the count to 1 (967ms)
    ✓ should have a starting count of 0 (98ms)
    ✓ should increment the count to 1 (928ms)
    ✓ should have a starting count of 0 (597ms)
    ✓ should increment the count to 1 (1304ms)


  12 passing (10s)
```

Approved!

**Pull request merged**